tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py

```diff
@@ -0,0 +1,158 @@
+import logging
+import pytest
+import time
+
+
+from ocs_ci.framework.pytest_customization.marks import tier4a
+from ocs_ci.framework import config
+from ocs_ci.ocs.dr.dr_workload import validate_data_integrity
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.node import wait_for_nodes_status, get_node_objs
+from ocs_ci.ocs.resources.pod import restart_pods_having_label
+from ocs_ci.helpers.dr_helpers import (
+    set_current_primary_cluster_context,
+    set_current_secondary_cluster_context,
+    get_current_primary_cluster_name,
+    get_active_acm_index,
+)
+from ocs_ci.framework.pytest_customization.marks import turquoise_squad
+from ocs_ci.utility import vsphere
+from ocs_ci.utility.utils import ceph_health_check
+
+logger = logging.getLogger(__name__)
+
+
+@tier4a
+@turquoise_squad
+class TestNoDataLossAndDataCorruptionOnFailures:
+    """
+    The  Objective of this test cases is to make sure that the MDR cluster remains accessible
+    and NO DU/DL/DC is observed when following Failures are induced with supported applications are running
+
+    1) Noobaa pods failures - repeat at least 5-7 times
+    2) Rolling reboot of the nodes in all zones one at a time
+    3) RHCS nodes failures
+        a. 1 RHCS node in one zone
+        b. All the RHCS nodes in one zone
+        c. All the RHCS nodes in one zone - Repeated to mimic Santa lucia issue
+
+    """
+
+    @pytest.mark.polarion_id("OCS-XXXX")
+    def test_no_data_loss_and_data_corruption_on_failures(
+        self, setup_acm_ui, nodes_multicluster, dr_workload
+    ):
+
+        # Deploy Subscription based application
+        sub = dr_workload(num_of_subscription=1)[0]
+        self.namespace = sub.workload_namespace
+        self.workload_type = sub.workload_type
+
+        # Deploy AppSet based application
+        appset = dr_workload(num_of_subscription=0, num_of_appset=1)[0]
+
+        # Workloads list
+        workloads = [sub, appset]
+
+        # Create application on Primary managed cluster
+        set_current_primary_cluster_context(self.namespace)
+        self.primary_cluster_name = get_current_primary_cluster_name(
+            namespace=self.namespace
+        )
+
+        # Validate data integrity
+        for wl in workloads:
+            config.switch_to_cluster_by_name(self.primary_cluster_name)
+            validate_data_integrity(wl.workload_namespace)
+
+        # Noobaa pod restarts atleast 5 times and verify the data integrity
+        restart_pods_having_label(label=constants.NOOBAA_APP_LABEL)
```

Comment:
IIUC, this line does restart only once. we need 5 times. Isn't loop missing here?

Reply:
yes, right. Added now